api/v1alpha1/thanosquery_types.go

```diff
@@ -23,22 +23,28 @@ import (
 // ThanosQuerySpec defines the desired state of ThanosQuery
 type ThanosQuerySpec struct {
 	CommonThanosFields `json:",inline"`
-
+	// Replicas is the number of router replicas.
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Required
+	Replicas int32 `json:"replicas,omitempty"`
+	// Labels are additional labels to add to the Querier component.
+	// +kubebuilder:validation:Optional
+	Labels map[string]string `json:"labels,omitempty"`
 	// Querier replica labels to configure.
-	// +optional
-	QuerierReplicaLabels []string `json:"replicaLabels,omitempty"`
-
+	// +kubebuilder:validation:Optional
+	QuerierReplicaLabels []string `json:"querierReplicaLabels,omitempty"`
 	// By default, the operator will add all discoverable StoreAPIs to the Querier,
 	// if they have store labels. You can optionally choose to override default
 	// StoreAPI selector labels, to select a subset of StoreAPIs to query.
-	// +optional
-	CustomStoreLabelSelector *metav1.LabelSelector `json:"storeLabelSelector,omitempty"`
+	// +kubebuilder:validation:Optional
+	CustomStoreLabelSelector *metav1.LabelSelector `json:"customStoreLabelSelector,omitempty"`
```

Comment:
some comments here

1. I prefer the former name `storeLabelSelector ` depending on the resolution of the comments below

2. should this be a list of selectors that are selected like  []{{x,y}, {a, b}} and the logic to select them looks like ( if (X AND Y ) || if (A AND B) ) - the reason being that you might want to match on receives and store gateways for example which will have different label sets

3. I think we can actually set the default values via kubebuilder here potentially but it might become complex and documentation being enough!